javacord-api/src/main/java/org/javacord/api/util/Specifiable.java

```diff
@@ -0,0 +1,45 @@
+package org.javacord.api.util;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This interface is used to provide a more stream lined way to interact with the instance as sub types.
+ * @param <S> Base instance type.
+ */
+public interface Specifiable<S> {
+
+    /**
+     * Get an {@link Optional} instance of an instance as specified type.
+     * If the instance is not castable to the specified type, the {@link Optional} will be empty.
+     * @param type Specified type, this instance will be tried to casted to.
+     * @param instance Instance which will be tried to get casted.
+     * @param <T> Specified type.
+     * @return Returns an {@link Optional} of the specified type if the instance is castable, if not it returns an empty
+     *         {@link Optional}.
+     * @throws NullPointerException Throws a {@link NullPointerException} if either {@code type} or {@code instance} is
+     *                              {@code null}.
+     * @see java.util.Optional
+     */
+    static <T> Optional<T> tryToSpecify(Class<T> type, Object instance) {
```

Comment:
`T` should also extend `S`
```suggestion
    static <T extends S> Optional<T> tryToSpecify(Class<T> type, Object instance) {
```